jira-clone/Toast.jsx

```diff
@@ -0,0 +1,51 @@
+// Source: https://github.com/oldboyxx/jira_clone/blob/26a9e77b1789fef9cb43edb5d6018cf1663cf035/client/src/App/Toast/index.jsx#L1
+import React, { useState, useEffect } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import pubsub from 'sweet-pubsub';
+import { uniqueId } from 'lodash';
+
+import { Container, StyledToast, CloseIcon, Title, Message } from './Styles';
+
+const Toast = () => {
+  const [toasts, setToasts] = useState([]);
+
+  useEffect(() => {
+    const addToast = ({ type = 'success', title, message, duration = 5 }) => {
+      const id = uniqueId('toast-');
+
+      setToasts(currentToasts => [...currentToasts, { id, type, title, message }]);
+
+      if (duration) {
+        setTimeout(() => removeToast(id), duration * 1000);
+      }
+    };
+
+    pubsub.on('toast', addToast);
+
+    return () => {
+      pubsub.off('toast', addToast);
+    };
+  }, []);
+
+  const removeToast = id => {
```

Comment:
> [!WARNING]
> It's a good idea to memoize this function's declaration using `useCallback`. It's especially appropriate to this function because it doesn't use any reactive/state values.